libs/community/tests/unit_tests/vectorstores/test_sqlserver.py

```diff
@@ -506,3 +517,92 @@ def test_that_case_sensitivity_does_not_affect_distance_strategy(
+        embedding_length=EMBEDDING_LENGTH,
+        # FakeEmbeddings returns embeddings of the same size as `embedding_length`.
+        embedding_function=FakeEmbeddings(size=EMBEDDING_LENGTH),
+        table_name=_TABLE_NAME,
+    )
+
+    # _provide_token is called only during Entra ID authentication.
+    provide_token.assert_called()
+
+
+# We need to mock this so that actual connection is not attempted
+# after mocking _provide_token.
+@mock.patch("sqlalchemy.dialects.mssql.dialect.initialize")
+@mock.patch(
+    "langchain_community.vectorstores.sqlserver.SQLServer_VectorStore._provide_token"
+)
+def test_that_given_a_connection_string_with_uid_and_pwd_entra_id_auth_is_not_used(
+    provide_token: Mock,
+    dialect_initialize: Mock,
+) -> None:
+    """Test that if a connection string is provided to SQLServer_VectorStore object,
+    and connection string has username and password, entra id authentication is not
+    used and connection is successful."""
+
+    # Connection string contains username and password,
+    # mssql+pyodbc://username:password@lc-test.database.windows.net,1433/lcvectorstore
+    # ?driver=ODBC+Driver+17+for+SQL+Server"
+    SQLServer_VectorStore(
+        connection_string=_CONNECTION_STRING_WITH_UID_AND_PWD,
+        embedding_length=EMBEDDING_LENGTH,
+        # FakeEmbeddings returns embeddings of the same size as `embedding_length`.
+        embedding_function=FakeEmbeddings(size=EMBEDDING_LENGTH),
+        table_name=_TABLE_NAME,
+    )
+
+    # _provide_token is called only during Entra ID authentication.
+    provide_token.assert_not_called()
+
+
+# We need to mock this so that actual connection is not attempted
+# after mocking _provide_token.
+@mock.patch("sqlalchemy.dialects.mssql.dialect.initialize")
+@mock.patch(
+    "langchain_community.vectorstores.sqlserver.SQLServer_VectorStore._provide_token"
+)
+def test_that_connection_string_with_trusted_connection_yes_does_not_use_entra_id_auth(
+    provide_token: Mock,
+    dialect_initialize: Mock,
+) -> None:
+    """Test that if a connection string is provided to SQLServer_VectorStore object,
+    and connection string has `trusted_connection` set to `yes`, entra id
+    authentication is not used and connection is successful."""
+
+    # Connection string does not contain username and password,
+    # but has `trusted_connection=yes`
+    # mssql+pyodbc://@lc-test.database.windows.net,1433/lcvectorstore
+    # ?driver=ODBC+Driver+17+for+SQL+Server&trusted_connection=yes"
+    SQLServer_VectorStore(
+        connection_string=_CONNECTION_STRING_WITH_TRUSTED_CONNECTION,
+        embedding_length=EMBEDDING_LENGTH,
+        # FakeEmbeddings returns embeddings of the same size as `embedding_length`.
+        embedding_function=FakeEmbeddings(size=EMBEDDING_LENGTH),
+        table_name=_TABLE_NAME,
+    )
+
+    # _provide_token is called only during Entra ID authentication.
+    provide_token.assert_not_called()
```

Comment:
mssql+pyodbc://@servername/dbname?driver=ODBC+Driver+17+for+SQL+Server --> Entra ID

This case is not tested

Reply:
This comment is resolved, but we did not have any new test. Can you confirm once Shade.

Reply:
The test ‘test_that_given_a_valid_entra_id_connection_string_entra_id_authentication_is_used’ has been updated to use different connection strings.

- a connection string with no parameters 
- a connection string with trusted_connection set to no.

Hence, the reason for resolving the comment.

Reply:
Thanks for confirming Shade.